src/crypto/sha256.cpp

```diff
@@ -484,6 +484,14 @@ void inline cpuid(uint32_t leaf, uint32_t subleaf, uint32_t& a, uint32_t& b, uin
 {
   __asm__ ("cpuid" : "=a"(a), "=b"(b), "=c"(c), "=d"(d) : "0"(leaf), "2"(subleaf));
 }
+
+/** Check whether the OS has enabled AVX registers. */
+bool AVXEnabled()
+{
+    uint32_t a, d;
+    __asm__("xgetbv" : "=a"(a), "=d"(d) : "c"(0));
```

Comment:
Possible to use `_xgetbv` intrinsic here? https://software.intel.com/en-us/node/694243

Reply:
I tried that, but it doesn't seem GCC has that.

Reply:
How about something like this?:
```
#ifndef _xgetbv
static inline unsigned long long _xgetbv(unsigned int index){
  unsigned int eax, edx;
  __asm__ __volatile__("xgetbv" : "=a"(eax), "=d"(edx) : "c"(index));
  return ((unsigned long long)edx << 32) | eax;
}
#endif
```
https://lists.macports.org/pipermail/macports-dev/2013-January/021782.html

Reply:
It's the same thing, just specialized for AVX (which AFAIK is the only thing it's useful for entirely).

Reply:
My GCC doesn't seem to have `<xsaveintrin.h>`. I'm going to leave this as-is.

Reply:
Access to `_xgetbv` requires compiling with `-mxsave`. That means creating a separate build library target just for systems with that options, etc. This is much more straightforward, and works on any recent x86 system.

Reply:
Agreed